### PR TITLE
Call bash with `/usr/bin/env bash` instead of `/bin/bash`

### PR DESCRIPTION
**Pull request by akshatagarwl**
On some systems `/bin/bash` is not available hence cilium doesn't compile. This changes almost all of the occurences of `/bin/bash` with `/usr/bin/env bash`

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml
Comment (qmonnet) [inline install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: Aren't you missing double quotes on this one?

File: test/k8sT/manifests/cilium-ds-clean-only.yaml
Comment (qmonnet) [inline test/k8sT/manifests/cilium-ds-clean-only.yaml]: Same here.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good overall, please fix the two occurrences below.

Could you also please squash your two commits together?

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml
Comment (akshatagarwl) [reply install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: Ohh my bad :facepalm:

**Review by kkourt — APPROVED** [review]
LGTM, thanks!

Was this done automatically via some script? If yes, as an optional change maybe you can add the command you used in the commit message?

**Review by qmonnet — APPROVED** [review]

**Comment by qmonnet** [comment]
test-me-please

**Review by tklauser — CHANGES_REQUESTED** [review]
It looks like the image lint CI job is currently failing for this PR: https://github.com/cilium/cilium/pull/16495/checks?check_run_id=2801445310

This is due to the changes in `images/`. You'll also need to update the `cilium-runtime` and `cilium-builder` images, see https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images for details.

**Comment by humancalico** [comment]
> Was this done automatically via some script? If yes, as an optional change maybe you can add the command you used in the commit message?

It was mostly done using VSCode's find and replace :sweat_smile:

**Comment by humancalico** [comment]
@cilium/build can you approve this build? https://github.com/cilium/cilium/actions/runs/926680439

**Comment by tklauser** [comment]
@humancalico sorry for the delay! I've now triggered the build. You should be able to update the image digest as soon as it ran through. Please ping me directly once you've pushed the updated commit with the new image digests. Thanks.

File: test/fuzzing/oss-fuzz-build.sh
Comment (kaworu) [inline test/fuzzing/oss-fuzz-build.sh]: This won't work:

```console
$ ./test/fuzzing/oss-fuzz-build.sh
/usr/bin/env: ‘bash -eu’: No such file or directory
/usr/bin/env: use -[v]S to pass options in shebang lines
```

```suggestion
#!/usr/bin/env bash

set -o errexit
set -o pipefail
set -o nounset
```

**Review by kaworu — CHANGES_REQUESTED** [review]

File: .github/workflows/build-images-ci.yaml
Comment (pchaigno) [inline .github/workflows/build-images-ci.yaml]: Why were the SHA256 hashes removed?

File: .github/workflows/build-images-ci.yaml
Comment (pchaigno) [inline .github/workflows/build-images-ci.yaml]: This GitHub workflow runs on `pull_request_target`. You will have to open a temporary, draft PR with that trigger changed to `pull_request` to properly test this pull request. Otherwise, it will run the master version of this workflow.

**Review by pchaigno — CHANGES_REQUESTED** [review]

File: .github/workflows/build-images-ci.yaml
Comment (akshatagarwl) [reply .github/workflows/build-images-ci.yaml]: Could this be an issue with the `update-cilium-runtime-image.sh` script?

File: test/fuzzing/oss-fuzz-build.sh
Comment (akshatagarwl) [reply test/fuzzing/oss-fuzz-build.sh]: Fixed. Thanks

File: .github/workflows/build-images-ci.yaml
Comment (akshatagarwl) [reply .github/workflows/build-images-ci.yaml]: @pchaigno I've opened #16543. Can you have a look and tell if that seems correct?

**Comment by humancalico** [comment]
@tklauser I've updated the image digests and opened a draft PR(#16543) to test as per the suggestion of @pchaigno.

File: .github/workflows/build-images-ci.yaml
Comment (tklauser) [reply .github/workflows/build-images-ci.yaml]: This is expected. The SHA256 hash will be updated once the image build action ran for a second time and `make update-runtime-image` is run again. We might need to clarify that bit in https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

Comment (pchaigno) [reply .github/workflows/build-images-ci.yaml]: @tklauser So they will be updated before we merge this PR, right?

Comment (tklauser) [reply .github/workflows/build-images-ci.yaml]: @pchaigno Yes, they will need to be updated using `make -C images update-runtime-image` before merging.

File: .github/workflows/build-images-ci.yaml
Comment (pchaigno) [reply .github/workflows/build-images-ci.yaml]: Looks good. I'll trigger the tests.

Comment (pchaigno) [reply .github/workflows/build-images-ci.yaml]: I forget that won't work with a forked repository :facepalm: I'll copy your branch and will send it on my side :-)

Comment (pchaigno) [reply .github/workflows/build-images-ci.yaml]: Testing at https://github.com/cilium/cilium/pull/16553.

Comment (pchaigno) [reply .github/workflows/build-images-ci.yaml]: The tests at #16553 are currently failing because of the missing SHA256 hashes. We'll have to rerun once those are updated.

**Review by kaworu — APPROVED** [review]

**Comment by tklauser** [comment]
@humancalico Thanks. The images have been built. Please run `make -C images/ update-runtime-image update-builder-image` again, amend the result and force push to the branch again. Then we should be good to go :rocket:

**Comment by humancalico** [comment]
Thanks @tklauser. Also should I squash the `update cilium-{runtime,builder}` commit with the `replace /bin/bash with /usr/bin/env bash` commit as well or keep them seperate?

**Comment by tklauser** [comment]
Thanks @humancalico. I think it is fine to keep the commits as they are, i.e. a separate commit updating the images.

**Review by tklauser — APPROVED** [review]

**Review by errordeveloper — APPROVED** [review]

**Comment by tklauser** [comment]
@humancalico It looks like another PR (#16470) was merged in the meantime, causing merge conflicts in this one. Could you please rebase your PR? Also, it looks like you'll need to run `make -C images update-cilium-runtime update-cilium-builder` once again, see https://github.com/cilium/cilium/pull/16495/checks?check_run_id=2856467904

**Comment by humancalico** [comment]
@tklauser I did a rebase and also ran `make -C images update-cilium-runtime update-cilium-builder` but the image digests didn't get updated

**Comment by pchaigno** [comment]
@humancalico What version of `docker buildx` are you running? Could you try rerunning `make -C images update-cilium-runtime update-cilium-builder` with the patch at https://github.com/cilium/cilium/pull/16621?

**Comment by humancalico** [comment]
> @humancalico What version of `docker buildx` are you running?

@pchaigno It seems like it is quite different from the PR you mentioned
```
github.com/docker/buildx v0.5.1-tp-docker 11057da37336192bfc57d81e02359ba7ba848e4a
```

> Could you try rerunning `make -C images update-cilium-runtime update-cilium-builder` with the patch at #16621?

Rebasing against that branch and running that command again didn't do anything either :(

**Comment by joestringer** [comment]
> > Could you try rerunning make -C images update-cilium-runtime update-cilium-builder with the patch at #16621?
> 
> Rebasing against that branch and running that command again didn't do anything either :(

You may want to consider swapping `FORCE_BUILD=1` for `FORCE_BUILD=0` (or vice-versa) in the runtime image dockerfile to force the dockerfile build to work correctly.

**Comment by joestringer** [comment]
@aanm Do you know the best way to get this to a point where we can merge it? I approved the branch deploy to build images, but it has failed. Seems like everyone is happy with the changes themselves, we just need to line up the build systems & get runtime/builder images built with these changes.

**Comment by humancalico** [comment]
@joestringer. Ran `make -C images/ update-runtime-image update-builder-image` once again and pushed the changes. I guess you would have to approve the build once again and I would have to update and push them one more time.

**Comment by aanm** [comment]
test-me-please

**Review by aanm — CHANGES_REQUESTED** [review]
@humancalico the test failures are legitimate:

```
Error: failed to start container "cilium-agent": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/usr/bin/env bash\": stat /usr/bin/env bash: no such file or directory": unknown              
```

**Comment by pchaigno** [comment]
> @aanm Do you know the best way to get this to a point where we can merge it? I approved the branch deploy to build images, but it has failed. Seems like everyone is happy with the changes themselves, we just need to line up the build systems & get runtime/builder images built with these changes.

We never succesfully tested the workflow changes. That needs to be done in a separate PR.

**Pull request closed without merging**
